### PR TITLE
Fix sharing icon

### LINKED ISSUE #40202: adjust icon of shared state in broadcrumb
> Shared
> image

I am not sure about this icon. Shouldn't we better use a different icon, e.g. this one?
image
https://fonts.google.com/icons?icon.style=Outlined&icon.set=Material+Symbols&selected=Material+Symbols+Outlined:folder_shared:FILL@0;wght@400;GRAD@0;opsz@24&icon.query=shar

_Originally posted by @szaimen in https://github.com/nextcloud/server/issues/40192#issuecomment-1702890351_

cc @nextcloud/designers

### PR DESCRIPTION
**Pull request by marcoambrosini**
* Resolves: #40202

## shared state
Screenshot from 2023-12-28 14-13-36

## unshared state
Screenshot from 2023-12-28 14-13-59

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by skjnldsv** [comment]
Seems like the first icon on your screenshot isn't what is being discussed on the mentioned issue: #40202
I fear we don't use that icon anywhere else :)
image

File: apps/files/src/views/FilesList.vue [hunk 1]
```diff
@@ -34,8 +34,8 @@
 						type="tertiary"
 						@click="openSharingSidebar">
 						<template #icon>
-							<LinkIcon v-if="shareButtonType === Type.SHARE_TYPE_LINK" />
-							<ShareVariantIcon v-else :size="20" />
+							<FolderAccountIcon v-if="shareButtonType !== null" />
```
Comment (skjnldsv) [inline apps/files/src/views/FilesList.vue:37]: @jancborchardt shall we lose the share link icon too?

**Comment by para28711** [comment]
Stephen f Balogh

**Comment by marcoambrosini** [comment]
> Seems like the first icon on your screenshot isn't what is being discussed on the mentioned issue: https://github.com/nextcloud/server/issues/40202

I see now that @jancborchardt refers to the folder with the link in it. I think we should use the folder-share material design icon more appropriate and I think that we should use that one in the files list too.

File: apps/files/src/views/FilesList.vue (hunk 1 above)
Comment (jancborchardt) [reply apps/files/src/views/FilesList.vue:37]: If it’s shared by link, that should be reflected?

**Comment by jancborchardt** [comment]
@marcoambrosini @skjnldsv let’s indeed focus on the issue at hand here to prevent creating deadlocks like these. :)

This icon https://fonts.google.com/icons?selected=Material%20Icons%3Aperson_add_alt_1%3A communicates both "Share" and "Add to share", so it can be used in both unshared and shared state, which we currently do not separate in the breadcrumbs either.

File: apps/files/src/views/FilesList.vue (hunk 1 above)
Comment (skjnldsv) [reply apps/files/src/views/FilesList.vue:37]: I agree, but this PR removes that

File: apps/files/src/views/FilesList.vue [hunk 2]
```diff
@@ -34,8 +34,8 @@
 						type="tertiary"
 						@click="openSharingSidebar">
 						<template #icon>
-							<LinkIcon v-if="shareButtonType === Type.SHARE_TYPE_LINK" />
```
Comment (szaimen) [inline apps/files/src/views/FilesList.vue:37]: this should probably be restored?

File: apps/files/src/views/FilesList.vue (hunk 1 above)
Comment (marcoambrosini) [reply apps/files/src/views/FilesList.vue:37]: @jancborchardt we said in yesterday's design review that we're not going to use this icon in iOS in favor of the native one. I think that in line with that, we should use the material folder-shared icon that @szaimen proposed in the issue both in the web interface and in the android apps.

**Comment by marcoambrosini** [comment]
> let’s indeed focus on the issue at hand 

The issue at hand proposes the usage of this new icon @jancborchardt 

> which we currently do not separate in the breadcrumbs either.

We currently have separate states in the breadcrumbs icon, just with a worse logic. A different icon is shown only if the folder is shared by link. No distinction is made between unshared and shared without link. 
I think it's just better to distinguish between shared and not shared.
Ref https://github.com/nextcloud/server/pull/40192

**Comment by jancborchardt** [comment]
So as @skjnldsv said, we don’t use the proposed icon anywhere. What we do use is these ones:
image
image

The core issue is mainly that in the breadcrumbs we still use the "triangle" share icon, and that needs to be replaced. If we discuss more than that then it will of course lead to unnecessary discussion and block a simple fix as this → so let’s keep the matters separate.

File: apps/files/src/views/FilesList.vue (hunk 2 above)
Comment (skjnldsv) [reply apps/files/src/views/FilesList.vue:37]: https://github.com/nextcloud/server/pull/42502#discussion_r1437444971

**Comment by jancborchardt** [comment]
Moving assignment from @marcoambrosini to @szaimen as discussed in the design team call. :)

**Comment by szaimen** [comment]
Superceded by https://github.com/nextcloud/server/pull/43433

**Pull request closed without merging**